resilience4j-commons-configuration/src/main/java/io/github/resilience4j/configuration/circuitbreaker/CircuitBreakerConfiguration.java

```diff
@@ -0,0 +1,79 @@
+package io.github.resilience4j.configuration.circuitbreaker;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreakerConfig;
+import org.apache.commons.configuration2.Configuration;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Optional;
+
+import static io.github.resilience4j.configuration.utils.ConfigurationUtil.getDuration;
+import static io.github.resilience4j.configuration.utils.ConfigurationUtil.getThrowableClassesByName;
+
+/**
+ * {@link CircuitBreakerConfiguration} is used to create {@link CircuitBreakerConfig} from a {@link Configuration}.
+ */
+public class CircuitBreakerConfiguration {
+    private static final String DEFAULT_NAME = "default";
+    private static final String DEFAULT_CONTEXT = "io.github.resilience4j.circuitbreaker";
+    private final Configuration config;
+
+    public CircuitBreakerConfiguration(final Configuration config) {
+        this(config, DEFAULT_CONTEXT);
+    }
+
+    public CircuitBreakerConfiguration(final Configuration config, final String circuitBreakerPrefix) {
+        this.config = config.subset(circuitBreakerPrefix);
+    }
+
+    public CircuitBreakerConfig get(String name, final CircuitBreakerConfig defaultConfig) {
+        final Configuration namedConfig = config.subset(name);
+
+        CircuitBreakerConfig baseConfig =
+            Optional.ofNullable(namedConfig.getString("baseConfig")).map(b -> {
+                return this.get(b, defaultConfig);
+            }).orElse(defaultConfig);
```

Comment:
There is minor concern to be aware of in the way that this is implemented since it is recursive, but I didn't want to over-engineer it if the maintainers are not concerned. Feedback on this welcome before I add similar logic to BulkheadConfiguration.